devito/passes/iet/engine.py

```diff
@@ -233,36 +239,37 @@ def abstract_efunc(efunc):
     return efunc
 
 
-def abstract_objects(objects):
+def abstract_objects(objects, sregistry=None):
     """
     Proxy for `abstract_object`.
     """
     # Precedence rules make it possible to reconstruct objects that depend on
     # higher priority objects
     priority = {
         DiscreteFunction: 1,
+        AbstractIncrDimension: 2,
+        BlockDimension: 3,
     }
```

Comment:
Not sure I understood this; could you help with a brief explanation?

Reply:
I remember I had a similar approach in my time-tiling, but was considered a kinda-hack :-)

Reply:
Thankfully GitHub preserves history!

Can you link me to that discussion please?

Reply:
Are there other places in the framework where something similar is being done?

Reply:
I'm not sure honestly, maybe? I mean, the concept of prioritizing processing is fairly widespread